### PR TITLE
Compile private C++20 modules

### PR DESCRIPTION
**Pull request by Arthapz**
We forgot to build privates modules

### CODE REVIEW

File: xmake/rules/c++/modules/modules_support/clang.lua [hunk 1]
```diff
@@ -149,6 +149,43 @@ function _build_modulefile(target, sourcefile, opt)
...
+        assert(compinst:compile(sourcefile, objectfile, {dependinfo = dependinfo, compflags = flags}))
+
+        -- update files and values to the dependent file
+        dependinfo.values = depvalues
+        table.join2(dependinfo.files, sourcefile)
+        depend.save(dependinfo, dependfile)
+    end
+end
+
+-- build module file
```
Comment (waruqi) [inline xmake/rules/c++/modules/modules_support/clang.lua:173]: incorrect comment?

File: xmake/rules/c++/modules/modules_support/clang.lua [hunk 2]
```diff
@@ -133,7 +133,7 @@ function _get_toolchain_includedirs_for_stlheaders(target, includedirs, clang)
     os.tryrm(tmpfile)
 end
 
--- build module file
+-- build interface module file
```
Comment (waruqi) [inline xmake/rules/c++/modules/modules_support/clang.lua:136]: and here?

File: xmake/rules/c++/modules/modules_support/msvc.lua [hunk 3]
```diff
@@ -84,7 +84,7 @@ function _add_objectfile_to_link_arguments(target, objectfile)
     common.localcache():save(cachekey)
 end
 
--- build module file
+-- build interface module file
```
Comment (waruqi) [inline xmake/rules/c++/modules/modules_support/msvc.lua:87]: here

**Comment by waruqi** [comment]
We also need to consider incremental compilation. It does not work for this test case. But other tests work.

```console
ruki-2:private_module ruki$ xmake 
[  0%]: generating.cxx.module.deps src/use.mpp
[  0%]: generating.cxx.module.deps src/dep1.mpp
[  0%]: generating.cxx.module.deps src/dep2.mpp
[ 14%]: generating.cxx.module.bmi dep1
[ 28%]: generating.cxx.module.bmi dep2
[ 42%]: build.cxx.module src/use.mpp
[ 71%]: archiving.release libprivate_module.a
[100%]: build ok!
ruki-2:private_module ruki$ xmake
[  0%]: generating.cxx.module.deps src/use.mpp
[ 42%]: build.cxx.module src/use.mpp
[ 71%]: archiving.release libprivate_module.a
[100%]: build ok!
```

**Comment by Arthapz** [comment]
> We also need to consider incremental compilation. It does not work for this test case. But other tests work.
> 
> for clang
> 
> ```
> ruki-2:private_module ruki$ xmake 
> [  0%]: generating.cxx.module.deps src/use.mpp
> [  0%]: generating.cxx.module.deps src/dep1.mpp
> [  0%]: generating.cxx.module.deps src/dep2.mpp
> [ 14%]: generating.cxx.module.bmi dep1
> [ 28%]: generating.cxx.module.bmi dep2
> [ 42%]: build.cxx.module src/use.mpp
> [ 71%]: archiving.release libprivate_module.a
> [100%]: build ok!
> ruki-2:private_module ruki$ xmake
> [  0%]: generating.cxx.module.deps src/use.mpp
> [ 42%]: build.cxx.module src/use.mpp
> [ 71%]: archiving.release libprivate_module.a
> [100%]: build ok!
> ```

now it should work

**Pull request merged**